web_api.py

```diff
@@ -0,0 +1,290 @@
+    # email = request.form.get('email')
+    # if email:
+    #     uid = save_upload_with_email(file, email)
+    # else:
+    #     new_filename = f"{filename}_{timestamp}_{uid}"
+    #     file_path = os.path.join(app.config['UPLOAD_FOLDER'], new_filename)
+    #     file.save(file_path)
+    # response = {'uid': uid}
+    # return jsonify(response)
+
+
+@app.route('/status/<uid>', methods=['GET'])
+def status(uid):
+    """
+    API endpoint for checking the status of an upload.
+
+    Args:
+        uid (str): The UID of the upload.
+
+    Returns:
+        flask.Response: JSON response containing the status, filename, timestamp, and explanation of the upload.
+    """
+    email = None
+    filename = None
+    print(uid)
+    if '@' in uid:
+        filename, email = uid.split(' ')
+        uid = None
+    # email = request.form.get('email')
+    # filename = request.form.get('filename')
+    if uid:
+        response = get_status_by_uid(uid)
+        if not response:
+            return jsonify({'status': 'not found'}), 404
+    elif email and filename:  # Only search if both
+        get_status_by_email_and_filename(email, filename)
+    else:
+        flash("Please enter a UID, or provide both email and filename")
+
+
+def generate_uid():
+    """
+    Generates a unique identifier (UID) for an upload.
+
+    Returns:
+        str: The generated UID.
+    """
+    return str(uuid.uuid4())
+
+
+def get_filename(filename):
+    """
+    Extracts the filename without the extension.
+
+    Args:
+        filename (str): The original filename.
+
+    Returns:
+        str: The filename without the extension.
+    """
+    return os.path.splitext(filename)[0]
+
+
+def find_file_by_uid(uid):
+    """
+    Finds the file path associated with a given UID.
+
+    Args:
+        uid (str): The UID of the upload.
+
+    Returns:
+        str: The file path if found, or None if not found.
+    """
+    for file in os.listdir(app.config['UPLOAD_FOLDER']):
+        if uid in file:
+            return os.path.join(app.config['UPLOAD_FOLDER'], file)
+    return None
+
+
+def get_status(file_path):
+    """
+    Determines the status of an upload based on the presence of an output file.
+
+    Args:
+        file_path (str): The file path of the upload.
+
+    Returns:
+        str: The status of the upload ('done' or 'pending').
+    """
+    output_file = get_output_file(file_path)
+    if output_file:
+        return 'done'
+    return 'pending'
+
+
+def get_file_details(file_path, file_status):
+    """
+    Extracts the filename, timestamp, and explanation from a file path and status.
+
+    Args:
+        file_path (str): The file path of the upload.
+        file_status (str): The status of the upload.
+
+    Returns:
+        Tuple[str, int, Optional[str]]: The filename, timestamp, and explanation.
+    """
+    filename_parts = os.path.basename(file_path).split('_')
+    filename = '_'.join(filename_parts[:-2])
```

Comment:
Assign constants to descriptive variables to enhance readability and simplify future changes ^__^ . Directly using them can lead to confusion and maintenance challenges.